sda/service.go

```diff
@@ -318,84 +303,69 @@ func (s *serviceManager) serviceByID(id ServiceID) (Service, bool) {
 	return serv, true
 }
 
-// ClientRequest is a generic packet to represent any kind of request a Service is
-// ready to process. It is simply a JSON packet containing two fields:
-// * Service: a string representing the name of the service for whom the packet
-// is intended for.
-// * Data: contains all the information of the request
-type ClientRequest struct {
-	// Name of the service to direct this request to
-	Service ServiceID
-	// Data containing all the information in the request
-	Data []byte
-}
-
-// ClientRequestID is the type that registered by the network library
-var ClientRequestID = network.RegisterPacketType(ClientRequest{})
-
-// CreateClientRequest creates a Request message out of any message that is
-// destined to a Service. XXX For the moment it uses protobuf, as it is already
-// handling abstract.Scalar/Public stuff that json can't do. Later we may want
-// to think on how to change that.
-func CreateClientRequest(service string, r interface{}) (*ClientRequest, error) {
-	sid := ServiceFactory.ServiceID(service)
-	log.Lvl1("Name", service, " <-> ServiceID", sid.String())
-	buff, err := network.MarshalRegisteredType(r)
-	if err != nil {
-		return nil, err
-	}
-	return &ClientRequest{
-		Service: sid,
-		Data:    buff,
-	}, nil
-}
-
 // Client is a struct used to communicate with a remote Service running on a
 // sda.Conode
 type Client struct {
-	ServiceID ServiceID
-	net       *network.Client
+	service string
 }
 
 // NewClient returns a client using the service s. It uses TCP communication by
 // default
 func NewClient(s string) *Client {
 	return &Client{
-		ServiceID: ServiceFactory.ServiceID(s),
-		net:       network.NewTCPClient(),
+		service: s,
 	}
 }
 
 // Send will marshal the message into a ClientRequest message and send it.
-func (c *Client) Send(dst *network.ServerIdentity, msg network.Body) (*network.Packet, error) {
-
-	m, err := network.NewNetworkPacket(msg)
+func (c *Client) Send(dst *network.ServerIdentity, path string, buf []byte) ([]byte, error) {
+	// Open connection to service.
+	url, err := getWebHost(dst)
 	if err != nil {
 		return nil, err
 	}
-
-	b, err := m.MarshalBinary()
+	log.Lvlf4("Sending %x to %s/%s/%s", buf, url, c.service, path)
+	ws, err := websocket.Dial(fmt.Sprintf("ws://%s/%s/%s", url, c.service, path),
+		"", "http://localhost/")
 	if err != nil {
 		return nil, err
 	}
+	if err = websocket.Message.Send(ws, buf); err != nil {
+		return nil, err
+	}
+	var rcv []byte
+	if err = websocket.Message.Receive(ws, &rcv); err != nil {
+		return nil, err
+	}
+	log.Lvlf4("Received %x", rcv)
+	return rcv, nil
+}
 
-	serviceReq := &ClientRequest{
-		Service: c.ServiceID,
-		Data:    b,
+// SendReflect wraps protobuf.(En|De)code to the Client.Send-function.
+func (c *Client) SendProtobuf(dst *network.ServerIdentity, msg interface{}, ret interface{}) error {
+	buf, err := protobuf.Encode(msg)
+	if err != nil {
+		return err
+	}
+	path := strings.Split(reflect.TypeOf(msg).String(), ".")[1]
```

Comment:
Would be better to put that as a one-line function to make sure we're doing the same everytime. (it's already present twice I believe)